src/js/utils.js

```diff
@@ -94,7 +92,7 @@ export function getDataAttributes(oPermutiveEl) {
 		return {};
 	}
 
-	return merge({}, ...Object.keys(oPermutiveEl.dataset)
+	return Object.assign({}, ...Object.keys(oPermutiveEl.dataset)
```

Comment:
It it possible to call map when Object.assign returns an object?

Reply:
https://puu.sh/DB1No/430e1cd4ca.png

Reply:
map is called on the `Object.keys()` which returns an array. I can split out the code so it's a bit clearer

Reply:
Alright, yeah it looks weird to do Object.assign and then mix an array and an empty object tbh

so then theorically we could do:

```jsx
return [...Object.keys(oPermutiveEl.dataset)].map(ect)
```

Reply:
The end goal is to return an object. So we're destructing the Object.keys array, which gives us a bunch of objects. And using `Obejct.assign` to merge those objects together. Essentially:

```
const keyValues = Object.keys(oPermutiveEl.dataset).map(etc);
// returns: [{ option1: value }, { option2: value }, etc]

return Object.assign({}, ...keyValues)
// copy all those properties onto a new object and return that object.
```

Reply:
I also think using `Objec.assign` in this case is the neatest approach.

The alternatives I can think of would be more complicate to read: (a) use `reduce` to merge every single key-value into the accumulator object, or (b) loop on the array with `for...of` to do something similar.